packages/commerce-sdk-react/src/test-utils.tsx

```diff
@@ -24,13 +24,18 @@ export const TEST_CONFIG = {
 }
 const TestProviders = (props: {
     children: React.ReactNode
-    optionalConfig?: Pick<CommerceApiProviderProps, 'headers' | 'fetchOptions'>
+    commerceApiProvider?: Partial<CommerceApiProviderProps>
+    queryClientProvider?: Partial<QueryClientProviderProps>
 }) => {
-    const queryClient = new QueryClient()
-    const {optionalConfig} = props
+    const {commerceApiProvider, queryClientProvider} = props
+    const queryClient = new QueryClient({
+        // During testing, we want things to fail immediately
+        defaultOptions: {queries: {retry: false}, mutations: {retry: false}},
+        ...queryClientProvider
```

Comment:
Hmm, if we want to use it like this, then the correct type for `queryClientProvider` should be `Partial<QueryClientConfig>` (and not `Partial<QueryClientProviderProps>`).

How about for now, let's remove the `queryClientProvider`? We can add it later, once there's a real need for it.

Reply:
ah I see what you meant before, reason I thought it was for the QueryClient is because most of the config will fall into there, not the provider itself. The Provider does not really have many configs that is useful in our case: https://react-query-v3.tanstack.com/reference/QueryClientProvider

Agree with you can remove it for now, and add it whenever it is needed